pkg/openstacknet/funcs.go

```diff
@@ -85,3 +91,173 @@ func GetOpenStackNetWithLabel(r common.ReconcilerCommon, namespace string, label
+) (map[string]ospdirectorv1beta1.OpenStackNet, error) {
+	osNetList, err := GetOpenStackNetsWithLabel(
+		r,
+		namespace,
+		labelSelector,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	osNetMap := map[string]ospdirectorv1beta1.OpenStackNet{}
+	for _, osNet := range osNetList.Items {
+		osNetMap[osNet.Spec.NameLower] = osNet
+	}
+
+	return osNetMap, nil
+}
+
+//
+// AddOSNetNameLowerLabels - add osnetcfg CR label reference which is used in
+// the in the osnetcfg controller to watch this resource and reconcile
+//
+func AddOSNetNameLowerLabels(
+	r common.ReconcilerCommon,
+	obj client.Object,
+	cond *ospdirectorv1beta1.Condition,
+	networkNameLowerNames []string,
+) error {
+
+	currentLabels := obj.GetLabels()
+	if currentLabels == nil {
+		currentLabels = map[string]string{}
+	}
+
+	osNetLabels := map[string]string{}
+	removedOsNets := map[string]bool{}
+	newOsNets := map[string]bool{}
+	networkNameLowerNamesMap := map[string]bool{}
+
+	for _, n := range networkNameLowerNames {
+		label := fmt.Sprintf("%s/%s", SubNetNameLabelSelector, n)
+		networkNameLowerNamesMap[label] = true
+	}
+
+	//
+	// get current osNet Labels and verify if nets got removed
+	//
+	for _, label := range reflect.ValueOf(currentLabels).MapKeys() {
+		//
+		// has label key SubNetNameLabelSelector string included?
+		//
+		if strings.HasSuffix(label.String(), SubNetNameLabelSelector) {
+			l := label.String()
+			osNetLabels[l] = currentLabels[l]
+
+			//
+			// if l is not in networkNameLowerNamesMap it got removed
+			//
+			if _, ok := networkNameLowerNamesMap[l]; !ok {
+				delete(currentLabels, l)
+				removedOsNets[l] = true
+
+			}
+		}
+	}
+
+	if len(newOsNets) > 0 {
+		common.LogForObject(
+			r,
+			fmt.Sprintf("%s %s removing network labels: %v",
+				obj.GetObjectKind().GroupVersionKind().Kind,
+				obj.GetName(),
+				removedOsNets,
+			),
+			obj,
+		)
+	}
+
+	//
+	// identify if nets got added
+	//
+	for label := range networkNameLowerNamesMap {
+		//
+		// if l is not in networkNameLowerNamesMap it got removed
+		//
+		if _, ok := osNetLabels[label]; !ok {
+			currentLabels[label] = strconv.FormatBool(true)
+			newOsNets[label] = true
+		}
+
+	}
+
+	if len(newOsNets) > 0 {
+		common.LogForObject(
+			r,
+			fmt.Sprintf("%s %s adding network labels: %v",
+				obj.GetObjectKind().GroupVersionKind().Kind,
+				obj.GetName(),
+				newOsNets,
+			),
+			obj,
+		)
+	}
+
+	//
+	// update labels on obj
+	//
+	_, err := controllerutil.CreateOrUpdate(context.TODO(), r.GetClient(), obj, func() error {
+		obj.SetLabels(
+			labels.Merge(
+				obj.GetLabels(),
+				currentLabels,
+			))
+
+		return nil
+	})
+	if err != nil {
+		cond.Message = fmt.Sprintf("Failed to update %s labels on %s", obj.GetObjectKind().GroupVersionKind().Kind, obj.GetName())
+		cond.Reason = ospdirectorv1beta1.ConditionReason(ospdirectorv1beta1.CommonCondReasonAddOSNetLabelError)
+		cond.Type = ospdirectorv1beta1.ConditionType(ospdirectorv1beta1.ConfigGeneratorCondTypeError)
```

Comment:
Is `ConfigGenerator` the right `TypeError` here?

Reply:
ah no, but was wondering if we should have generic Types as we have Error,Creating, Created,... on all of the objects. maybe they could just use all CommonType, like we already have CommonCondTypeError? We likely never change an Error to be named different between the CRDs?

Reply:
That seems reasonable to me

Reply:
done for this one